### PR TITLE
Fix multiple local network permission alert requesting [APP-190] [PLAN-280]

### PR DESCRIPTION
**Pull request by MaiklKasperovich**
# What was changed?

Fix multiple local network permission alert requesting

# Checklist:

- [x] Title of the PR contains ticket link as [APP-XXX], [PLAN-XXX] 
- [x] I've ran `yarn lint` to validate code style. (`yarn lint:fix` to automatically fix issues)
- [x] I have performed a self-review of my code in Pull Request
- [ ] I've checked Android and iOS platforms, attached screenshots/videos for UI changes
- [ ] I squashed commits after review before merging my ticket and rebased to fix conflicts ([first squash with interactive rebase then rebase on top and fix conflicts then git push --force-with-lease](https://www.loom.com/share/b98a534574c642b3ace78646b07f7375))
- [ ] I have added Unit tests for bugfix

# Screenshots and videos

https://github.com/user-attachments/assets/cc51badd-9819-45d1-9510-833dc51610a3

# Related
 - https://jira.hqsoftwarelab.net/browse/PLAN-280
 - https://planitar.atlassian.net/browse/APP-190

### CODE REVIEW

File: ios/LocalNetworkPrivacy/LocalNetworkPrivacy.m [hunk 1]
```diff
@@ -26,22 +28,62 @@ - (void)dealloc {
...
+    BOOL permissionRequestedBefore = [[NSUserDefaults standardUserDefaults] boolForKey:kLocalNetworkPermissionRequestedKey];
 
-    self.timer = [NSTimer scheduledTimerWithTimeInterval:2 repeats:NO block:^(NSTimer * _Nonnull timer) {
-        [self.timer invalidate];
-        self.completion(NO);
-    }];
+    if (!permissionRequestedBefore) {
+        [self logPermissionStatus];
+        [[NSUserDefaults standardUserDefaults] setBool:YES forKey:kLocalNetworkPermissionRequestedKey];
+        [[NSUserDefaults standardUserDefaults] synchronize];
```
Comment (jegora) [inline ios/LocalNetworkPrivacy/LocalNetworkPrivacy.m]: `synchronize` is no longer necessary to use: https://developer.apple.com/documentation/macos-release-notes/foundation-release-notes?language=objc

File: ios/LocalNetworkPrivacy/LocalNetworkPrivacy.m [hunk 2]
```diff
@@ -26,22 +28,62 @@ - (void)dealloc {
...
+        [self.service publish];
+
+        self.timer = [NSTimer scheduledTimerWithTimeInterval:0.5 repeats:YES block:^(NSTimer * _Nonnull timer) {
+            if (self.publishing) {
+                NSLog(@"Local network permission status: Requesting");
+            } else {
+                [self.timer invalidate];
+                if (self.completion) {
+                    self.completion(self.service.includesPeerToPeer);
+                }
```
Comment (jegora) [inline ios/LocalNetworkPrivacy/LocalNetworkPrivacy.m]: create a separate function for completion

File: ios/LocalNetworkPrivacy/LocalNetworkPrivacy.m [hunk 3]
```diff
@@ -26,22 +28,62 @@ - (void)dealloc {
...
+            if (self.completion) {
+                self.completion(NO);
+            }
+        }];
+    }
 }
 
+- (void)logPermissionStatus {
+    NSLog(@"Local network permission status: Requesting");
+}
```
Comment (jegora) [inline ios/LocalNetworkPrivacy/LocalNetworkPrivacy.m]: is logging still needed?

File: ios/LocalNetworkPrivacy/LocalNetworkPrivacy.m [hunk 4]
```diff
@@ -26,22 +28,62 @@ - (void)dealloc {
...
+        [[NSUserDefaults standardUserDefaults] setBool:YES forKey:kLocalNetworkPermissionRequestedKey];
+        [[NSUserDefaults standardUserDefaults] synchronize];
+
+        self.publishing = YES;
+        self.service.delegate = self;
+        [self.service publish];
+
+        self.timer = [NSTimer scheduledTimerWithTimeInterval:0.5 repeats:YES block:^(NSTimer * _Nonnull timer) {
+            if (self.publishing) {
+                NSLog(@"Local network permission status: Requesting");
```
Comment (jegora) [inline ios/LocalNetworkPrivacy/LocalNetworkPrivacy.m]: remove this if this is not needed

**Comment by jegora** [comment]
form this PR according to all the rules (description, related ticket refs, checklist etc.)

**Review by jegora — CHANGES_REQUESTED** [review]
form this PR according to all the rules (description, related ticket refs, checklist etc.)

File: ios/LocalNetworkPrivacy/LocalNetworkPrivacy.m (hunk 3 above)
Comment (MaiklKasperovich) [reply ios/LocalNetworkPrivacy/LocalNetworkPrivacy.m]: removed

File: ios/LocalNetworkPrivacy/LocalNetworkPrivacy.m (hunk 2 above)
Comment (MaiklKasperovich) [reply ios/LocalNetworkPrivacy/LocalNetworkPrivacy.m]: done

File: ios/LocalNetworkPrivacy/LocalNetworkPrivacy.m (hunk 1 above)
Comment (MaiklKasperovich) [reply ios/LocalNetworkPrivacy/LocalNetworkPrivacy.m]: removed

**Comment by MaiklKasperovich** [comment]
> form this PR according to all the rules (description, related ticket refs, checklist etc.)

done

**Review by jegora — APPROVED** [review]

**Pull request merged**